great_expectations/core/expectation_suite.py

```diff
@@ -669,29 +672,204 @@ def add_expectation(
-        return [e for e in self.expectations if "column" in e.kwargs]
+        expectation_configurations: List[ExpectationConfiguration] = list(
+            filter(
+                lambda element: element.get_domain_type() == MetricDomainTypes.COLUMN,
+                self.expectations,
+            )
+        )
 
+        expectation_configuration: ExpectationConfiguration
+        kwargs: dict
+        column_name: str
+        for expectation_configuration in expectation_configurations:
+            kwargs = deep_filter_properties_iterable(
+                properties=expectation_configuration.kwargs, clean_falsy=True
+            )
+            column_name = kwargs.pop("column")
+            expectation_configuration.kwargs = {"column": column_name, **kwargs}
+
+        return expectation_configurations
+
+    # noinspection PyPep8Naming
     def get_column_pair_expectations(self) -> List[ExpectationConfiguration]:
         """Return a list of column_pair map expectations."""
-        return [
-            e
-            for e in self.expectations
-            if "column_A" in e.kwargs and "column_B" in e.kwargs
-        ]
+        expectation_configurations: List[ExpectationConfiguration] = list(
+            filter(
+                lambda element: element.get_domain_type()
+                == MetricDomainTypes.COLUMN_PAIR,
+                self.expectations,
+            )
+        )
+
+        expectation_configuration: ExpectationConfiguration
+        kwargs: dict
+        column_A_name: str
+        column_B_name: str
+        for expectation_configuration in expectation_configurations:
+            kwargs = deep_filter_properties_iterable(
+                properties=expectation_configuration.kwargs, clean_falsy=True
+            )
+            column_A_name = kwargs.pop("column_A")
+            column_B_name = kwargs.pop("column_B")
+            expectation_configuration.kwargs = {
+                "column_A": column_A_name,
+                "column_B": column_B_name,
+                **kwargs,
+            }
+
+        return expectation_configurations
 
     def get_multicolumn_expectations(self) -> List[ExpectationConfiguration]:
         """Return a list of multicolumn map expectations."""
-        return [e for e in self.expectations if "column_list" in e.kwargs]
+        expectation_configurations: List[ExpectationConfiguration] = list(
+            filter(
+                lambda element: element.get_domain_type()
+                == MetricDomainTypes.MULTICOLUMN,
+                self.expectations,
+            )
+        )
+
+        expectation_configuration: ExpectationConfiguration
+        kwargs: dict
+        column_list: str
+        for expectation_configuration in expectation_configurations:
+            kwargs = deep_filter_properties_iterable(
+                properties=expectation_configuration.kwargs, clean_falsy=True
+            )
+            column_list = kwargs.pop("column_list")
```

Comment:
Is this guaranteed to be in kwargs? If not, we should be careful of using `pop` without a second arg.

Reply:
@cdkini Guaranteed.

Reply:
Similar to my above comment about popping.